src/main/java/uk/ac/sanger/sccp/stan/service/history/HistoryServiceImp.java

```diff
@@ -350,13 +416,15 @@ public String resultDetail(ResultOp result, Map<Integer, Slot> slotIdMap) {
     /**
      * Creates history entries for the given operations, where relevant to the specified samples
      * @param operations the operations to represent in the history
-     * @param sampleIds the ids of relevant samples
+     * @param sampleIds the ids of relevant samples, or null to not filter by sample
      * @param labware the relevant labware
      * @param opWork a map of op id to work numbers
+     * @param singleWorkNumber a single work number applicable to all operations
      * @return a list history entries for the given operations
      */
     public List<HistoryEntry> createEntriesForOps(Collection<Operation> operations, Set<Integer> sampleIds,
-                                                  Collection<Labware> labware, Map<Integer, Set<String>> opWork) {
+                                                  Collection<Labware> labware, Map<Integer, Set<String>> opWork,
+                                                  String singleWorkNumber) {
```

Comment:
I have a doubt. In the following methods, why we have an argument for a single workNumber and another for a list of worknumbers?

Reply:
If you are looking up the history for a particular work number, then all the entries created will list the same work number. (In that case, opWork is null and singleWorkNumber is a string.) If you are looking up the history for something else, like a sample, then the different operations might be associated with different work numbers, so we need to look up the correct work numbers for each operation. (In that case, opWork will be a map from operation ids to work numbers, and singleWorkNumber will be null.)